src/YAMLValidator/Validator.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Json.Schema;
+using Yaml2JsonNode;
+// using System.Text.Json;
+using System.Text.Json.Nodes;
+namespace Microsoft.PowerPlatform.PowerApps.Persistence;
+
+internal sealed class Validator
+{
+    public JsonSchema? Schema { get; set; }
+    public string? Yaml { get; set; }
+
+    private readonly EvaluationOptions _verbosityOptions;
+    // private readonly JsonSerializerOptions _serializerOptions;
+
+
+    public Validator()
+    {
+        // to do: add verbosity flag and allow users to choose verbosity of evaluation
+        _verbosityOptions = new EvaluationOptions
+        {
+            OutputFormat = OutputFormat.Hierarchical
+        };
+        // _serializerOptions = new JsonSerializerOptions
+        // {
+        //     Converters = { new EvaluationResultsJsonConverter() }
+        // };
+    }
+
+    public bool Validate()
```

Comment:
quick update:
- I did[ fix this](https://github.com/microsoft/PowerApps-Tooling/blob/919ed6c42f75bc60cc6f1856ed66698cd28fdcba/src/YAMLValidator/Validator.cs), but I want new feedback, I encapsulated the returned results from the ```.evaluate function``` which produces a EvaluationResults object which has quite a few fields that may be too verbose. I know it goes against your advice of making the application stateless, but it was convenient for testing and outputting errors.

- Testing: since the encapsulated object also has a field indicating whether the whole schema matched or not I just use this for the updated tests

- OneOf edge case for false schemas: testing can be made more robust for false cases by actually looking at specific errors [here](https://github.com/microsoft/PowerApps-Tooling/blob/919ed6c42f75bc60cc6f1856ed66698cd28fdcba/src/YAMLValidator/YamlValidatorResults.cs) and validating them in tests instead of just looking at schema true or false. I want to do this in another PR though and keep the focus on this one on simple functionality testing of the validation. thoughts?